frontend/src/components/pages/components/collab-editor/collab-editor.tsx

```diff
@@ -0,0 +1,173 @@
+      },
+      toolbar: [
+        [{ 'header': [1, 2, 3, 4, 5, 6, false] }],
+        ['bold', 'italic', 'strike'],
+        [{ 'list': 'ordered' }, { 'list': 'bullet' }],
+        ['blockquote', 'code-block'],
+        ['image', 'link'],
+      ],
+    },
+    formats: [
+      'header',
+      'font',
+      'size',
+      'bold',
+      'italic',
+      'strike',
+      'blockquote',
+      'list',
+      'bullet',
+      'indent',
+      'link',
+      'image',
+      'code-block',
+    ],
+  });
+
+  const onSave = (): void => {
+    const turndownService = new TurndownService();
+    const markdown = turndownService.turndown(contentInput);
+    handleSaveConfirm(titleInput, markdown);
+  };
+
+  const onCancel = (): void => {
+    handleCancel();
+  };
+
+  useEffect(() => {
+    if (quill) {
+      const pagePath = replaceIdParam(AppRoute.CONTENT_SETTING, paramsId || '');
+      const ydoc = new Y.Doc();
+      const provider = new WebrtcProvider(`wss://${url}${pagePath}`, ydoc);
+
+      const yQuillTextYtype = ydoc.getText('quill');
+      new QuillBinding(yQuillTextYtype, quill, provider?.awareness);
+
+      provider.awareness.setLocalStateField('user', {
+        name: userName,
+        color: 'blue',
+      });
+
+      provider.awareness.setLocalStateField('quillSettings', {
+        isInputSet: false,
+      });
+
+      provider.awareness.getStates().forEach((state) => {
+        if (!state.quillSettings.isInputSet) {
+          const md = new MarkdownIt();
+          const result = md.render(content || '');
+          quill.clipboard.dangerouslyPasteHTML(result);
+
+          provider.awareness.setLocalStateField('quillSettings', {
+            isInputSet: true,
+          });
+        }
+      });
+
+      quill.on('text-change', () => {
+        const html = quill.root.innerHTML;
+        setContentInput(html);
+      });
+
+      return (): void => {
+        provider.destroy();
+        ydoc.destroy();
+      };
+    }
+  }, [quill]);
+
+  const onInputChange = ({
+    target,
+  }: React.ChangeEvent<HTMLInputElement>): void => {
+    setTitleInput(target.value);
+  };
+
+  return (
+    <>
+      <Row className="mb-4">
+        <Col className="d-flex justify-content-between">
+          <InputGroup>
+            <FormControl value={titleInput} onChange={onInputChange} />
+          </InputGroup>
+        </Col>
+      </Row>
+      <Row className="mb-4">
+        <Col>
+          <Card border="light" className={getAllowedClasses(styles.content)}>
+            <div className={getAllowedClasses(styles.editor)}>
+              <div ref={quillRef} />
+            </div>
+          </Card>
+        </Col>
+      </Row>
+      <Row className="mb-4">
+        <Col>
+          <Button onClick={onSave} variant="success" size="sm" className="me-3">
+            Save
+          </Button>
+          <Button onClick={onCancel} variant="warning" size="sm">
+            Cancel
+          </Button>
```

Comment:
You can invoke on `handleCancel` right on the element:

```suggestion
          <Button onClick={handleCancel} variant="warning" size="sm">
            Cancel
          </Button>
```